### PR TITLE
refactor: modernise the validation in active code and the associated tests for dealing with integer arrays

### PR DESCRIPTION
**Pull request by martin-georgiev**
## Summary by CodeRabbit

- **Refactor**
  - Improved data type consistency in numeric processing by switching from string representations to native integer values.
  - Enhanced error handling to deliver clear, targeted feedback when invalid numeric inputs are encountered.

- **New Features**
  - Introduced new, specialized error messages to indicate specific validation issues with numeric data.

- **Tests**
  - Expanded test coverage to include additional scenarios for invalid entries, boundary cases, and out-of-range values.
  - Streamlined test setups for more reliable validation of numeric transformations.

### CODE REVIEW

File: src/MartinGeorgiev/Doctrine/DBAL/Types/BaseIntegerArray.php [hunk 1]
```diff
@@ -14,45 +15,74 @@
...
+
+        $doesNotFitIntoPHPInteger = $stringValue !== (string) (int) $stringValue;
+        if ($doesNotFitIntoPHPInteger) {
+            throw InvalidIntegerArrayItemForPHPException::forValueOutOfRangeInPHP($item, static::TYPE_NAME);
+        }
+
+        $doesNotFitIntoDatabaseInteger = $stringValue < $this->getMinValue() || $stringValue > $this->getMaxValue();
+        if ($doesNotFitIntoDatabaseInteger) {
+            throw InvalidIntegerArrayItemForPHPException::forValueOutOfRangeInDatabaseType($item, static::TYPE_NAME);
         }
```
[the remark this answers is not in the document]
Comment (martin-georgiev) [reply src/MartinGeorgiev/Doctrine/DBAL/Types/BaseIntegerArray.php]: Good catch. Changes applied.
